libraries/AP_HAL_ChibiOS/hwdef/AET-H743-Basic/hwdef.dat

```diff
@@ -0,0 +1,216 @@
+# USART3
+PD9 USART3_RX USART3
+PD8 USART3_TX USART3
+
+# UART4
+PB9 UART4_TX UART4
+PB8 UART4_RX UART4
+
+# USART6
+PC7 TIM3_CH2 TIM3 RCININT PULLDOWN LOW
+PC6 USART6_TX USART6 NODMA
+
+# as an alternative config setup the RX6 pin as a uart. This allows
+# for bi-directional UART based receiver protocols such as FPort
+# without any extra hardware
+PC7 USART6_RX USART6 NODMA ALT(1)
+
+# UART7
+PE7 UART7_RX UART7
+PE8 UART7_TX UART7
+PE10 UART7_CTS UART7
+PE9  UART7_RTS UART7
+
+# UART8
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# Serial functions
+define DEFAULT_SERIAL1_BAUD 115
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_MAVLink2
+define DEFAULT_SERIAL2_BAUD 230
+define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_GPS
+define DEFAULT_SERIAL3_BAUD 115
+define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_MAVLink2
+define DEFAULT_SERIAL4_BAUD 230
+define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_GPS
+define DEFAULT_SERIAL6_BAUD 115
+define DEFAULT_SERIAL6_PROTOCOL SerialProtocol_RCIN
+define DEFAULT_SERIAL7_BAUD 115
+define DEFAULT_SERIAL7_PROTOCOL SerialProtocol_MAVLink2
+define DEFAULT_SERIAL8_BAUD 115
+define DEFAULT_SERIAL8_PROTOCOL SerialProtocol_MAVLink2
+
+# CAN bus
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+PD3 GPIO_CAN1_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(70)
+
+# Motors
+PB0  TIM8_CH2N TIM8  PWM(1)  GPIO(50)
+PB1  TIM8_CH3N TIM8  PWM(2)  GPIO(51)
+PA0  TIM5_CH1  TIM5  PWM(3)  GPIO(52)
+PA1  TIM5_CH2  TIM5  PWM(4)  GPIO(53)
+PA2  TIM5_CH3  TIM5  PWM(5)  GPIO(54)
+PA3  TIM5_CH4  TIM5  PWM(6)  GPIO(55)
+PD12 TIM4_CH1 TIM4 PWM(7) GPIO(56)
+PD13 TIM4_CH2 TIM4 PWM(8) GPIO(57)
+PD14 TIM4_CH3 TIM4 PWM(9) GPIO(58)
+PD15 TIM4_CH4 TIM4 PWM(10) GPIO(59)
+PE5  TIM15_CH1 TIM15 PWM(11) GPIO(60)
+PE6  TIM15_CH2 TIM15 PWM(12) GPIO(61)
+PA8 TIM1_CH1 TIM1 PWM(13) GPIO(62) # for WS2812 LED
+
+# Beeper
+PA15 TIM2_CH1 TIM2 GPIO(32) ALARM
+
+# microSD support
+PC8 SDMMC1_D0 SDMMC1
+PC9 SDMMC1_D1 SDMMC1
+PC10 SDMMC1_D2 SDMMC1
+PC11 SDMMC1_D3 SDMMC1
+PC12 SDMMC1_CK SDMMC1
+PD2 SDMMC1_CMD SDMMC1
+
+# GPIOs
+PD10  PINIO1 OUTPUT GPIO(81) LOW
+PD11  PINIO2 OUTPUT GPIO(82) LOW
+
+DMA_PRIORITY S*
+
+define HAL_STORAGE_SIZE 32768
+
+# use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+STORAGE_FLASH_PAGE 14
+
+# spi devices
+SPIDEV icm42688_0  SPI1 DEVID1 IMU1_CS      MODE3  2*MHZ 16*MHZ   # Clock is 100Mhz so highest clock <= 24Mhz is 100Mhz/8
+SPIDEV icm42688_1  SPI4 DEVID1 IMU3_CS      MODE3  2*MHZ 16*MHZ
+SPIDEV osd         SPI2 DEVID4 MAX7456_CS   MODE0 10*MHZ 10*MHZ
+
+
+DMA_NOSHARE SPI1* SPI4*
+
+# no built-in compass, but probe the i2c bus for all possible
+# external compass types
+define ALLOW_ARM_NO_COMPASS
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+define HAL_I2C_INTERNAL_MASK 0
+define HAL_COMPASS_AUTO_ROT_DEFAULT 2
+
+# two IMUs
+# ICM42688P, ICM42688P
+IMU Invensensev3 SPI:icm42688_0 ROTATION_YAW_180
+IMU Invensensev3 SPI:icm42688_1 ROTATION_YAW_270
+define HAL_DEFAULT_INS_FAST_SAMPLE 1
+
+# BAROs
+BARO SPL06  I2C:0:0x76
+BARO MS56XX I2C:0:0x77
```

Comment:
Does it actually ship with all of these?

Reply:
No, we may have several different BOMs, some with SPL06 and some DSP310. Others will be deleted.